src/services/expense.services.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+let expenses = [];
+
+const reset = () => {
+  expenses = [];
+};
+
+const getAll = (data) => {
+  const { userId, categories, from, to } = data;
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === Number(userId),
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.category === categories,
+    );
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses.filter((expense) => {
+      return new Date(expense.spentAt) >= new Date(from);
+    });
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses.filter((expense) => {
+      return new Date(expense.spentAt) <= new Date(to);
+    });
+  }
```

Comment:
you use filter at least three times here. would be good to combine it to iterate only once